components/file_system/src/thread_io.rs

```diff
@@ -0,0 +1,269 @@
+
+        io_bytes
+    }
+}
+
+impl Default for AtomicIOBytes {
+    fn default() -> Self {
+        let bytes = IOBytes::default();
+
+        AtomicIOBytes {
+            read: AtomicU64::new(bytes.read),
+            write: AtomicU64::new(bytes.write),
+        }
+    }
+}
+
+impl AtomicIOBytes {
+    fn load(&self, order: Ordering) -> IOBytes {
+        IOBytes {
+            read: self.read.load(order),
+            write: self.write.load(order),
+        }
+    }
+
+    fn store(&self, val: IOBytes, order: Ordering) {
+        self.read.store(val.read, order);
+        self.write.store(val.write, order);
+    }
+
+    fn fetch_add(&self, other: IOBytes, order: Ordering) {
+        self.read.fetch_add(other.read, order);
+        self.write.fetch_add(other.write, order);
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use libc::O_DIRECT;
+    use maligned::{AsBytes, AsBytesMut, A512};
+    use std::{
+        fs::OpenOptions,
+        io::{Read, Write},
+        os::unix::prelude::OpenOptionsExt,
+    };
+    use tempfile::{tempdir, tempdir_in};
+
+    use super::*;
+
+    #[test]
+    fn test_read_bytes() {
+        let tmp = tempdir_in("/var/tmp").unwrap_or_else(|_| tempdir().unwrap());
+        let file_path = tmp.path().join("test_read_bytes.txt");
+        {
+            let mut f = OpenOptions::new()
+                .write(true)
+                .create(true)
+                .custom_flags(O_DIRECT)
+                .open(&file_path)
+                .unwrap();
+            let w = vec![A512::default(); 10];
+            f.write_all(w.as_bytes()).unwrap();
+            f.sync_all().unwrap();
+        }
+        let mut f = OpenOptions::new()
+            .read(true)
+            .custom_flags(O_DIRECT)
+            .open(&file_path)
+            .unwrap();
+        let mut w = vec![A512::default(); 1];
+        let origin_io_bytes = fetch_thread_io_bytes(IOType::Other);
+
+        for i in 1..=10 {
+            f.read_exact(w.as_bytes_mut()).unwrap();
+
+            let io_bytes = fetch_thread_io_bytes(IOType::Other);
+
+            assert_eq!(i * 512 + origin_io_bytes.read, io_bytes.read);
+        }
+    }
+
+    #[test]
+    fn test_write_bytes() {
+        let tmp = tempdir_in("/var/tmp").unwrap_or_else(|_| tempdir().unwrap());
+        let file_path = tmp.path().join("test_write_bytes.txt");
+        let mut f = OpenOptions::new()
+            .write(true)
+            .create(true)
+            .custom_flags(O_DIRECT)
+            .open(&file_path)
+            .unwrap();
+
+        let w = vec![A512::default(); 1];
+
+        let origin_io_bytes = fetch_thread_io_bytes(IOType::Other);
+        for i in 1..=10 {
+            f.write_all(w.as_bytes()).unwrap();
+            f.sync_all().unwrap();
+
+            let io_bytes = fetch_thread_io_bytes(IOType::Other);
+
+            assert_eq!(i * 512 + origin_io_bytes.write, io_bytes.write);
+        }
+    }
+
+    #[bench]
+    fn bench_fetch_thread_io_bytes(b: &mut test::Bencher) {
```

Comment:
Re-run the bench and paste the results here as comments.

Reply:
test thread_io::tests::bench_fetch_thread_io_bytes ... bench:       5,488 ns/iter (+/- 608)

Reply:
Actually, I meant pasting the result as code comments😂

Reply:
Oh, my mistake😂